core/server/api/v2/utils/validators/utils/json-schema.js

```diff
@@ -3,15 +3,22 @@ const Ajv = require('ajv');
 const stripKeyword = require('./strip-keyword');
 const common = require('../../../../../lib/common');
 
-const validate = (schema, definitions, data) => {
-    const ajv = new Ajv({
-        allErrors: true,
-        useDefaults: true
-    });
+const ajv = new Ajv({
+    allErrors: true,
+    useDefaults: true
+});
 
-    stripKeyword(ajv);
+stripKeyword(ajv);
 
-    const validation = ajv.addSchema(definitions).compile(schema);
+const getValidation = (schema, def) => {
+    if (!ajv.getSchema(def.$id)) {
```

Comment:
Was this needed? From what I understood from the acv docs it already uses a stable JSON stringify on the schema as a key for the compiled validator cache - or does that only apply to `.compile()` rather than `.addSchema()`?

Reply:
From what I understand `addSchema` doesn't compile the schema, it's used for definitions or dependencies that are used in compiled schemas.

We have to add the definitions at least once, and if you try to add the same definition twice, it will throw an error. So I _think_ this is needed :sweat_smile: